### ISSUE TITLE
Use noindex for paginated pages/posts is being applied to archives

### ISSUE THREAD

**Issue by wpsmort** [semperfiwebdesign/all-in-one-seo-pack#906]
The Use noindex for paginated pages/posts option should only apply to posts and pages, but it's being applied incorrectly to categories, tags and other archives.

Need to test to see if this is happening for Use nofollow for paginated pages/posts and Remove Descriptions For Paginated Pages as well.

The code is a jumbled mess and needs to be cleaned up.  We also need to improve the filter hooks and documentation.

**Comment by wpsmort** [comment]
I have tested this and confirmed that all three settings apply to all paginated content including pages, posts, CPTs, blog page, category, tag, taxonomy, date and author archives.

The three settings should only apply to pages, posts and CPTs.   They should not apply to archives.

We need to make the following changes:

1). Change the behavior of the three settings so they only apply to pages, posts and CPTs
2). Add a filter for noindex paginated content that can be used with any conditional tag
3). Add a filter for nofollow paginated content that can be used with any conditional tag
4). Add a filter for remove descriptions for paginated content that can be used with any conditional tag

Once this is done we should update the help text and documentation to reflect that these settings only apply to pages, posts, and CPTS, add new documentation for the filters and open a new issue for trimming these three options.

**Comment by arnaudbroes** [comment]
@wpsmort I've fixed (1). I don't think additional filters are needed here since we have the aioseop_robots_meta filter which can be used to filter the robots meta tag value. Our documentation was missing but I added it about a week ago - https://semperplugins.com/documentation/aioseop-robots-meta/
